components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+                            // Get the subject claim in OIDC dialect.
+                            String userIdClaimUriInLocalDialect = claimMapping.getLocalClaim().getClaimUri();
+                            userIdClaimUriInOIDCDialect =
+                                    getUserIdClaimUriInOIDCDialect(userIdClaimUriInLocalDialect, spTenantDomain);
+                            break;
+                        }
+                    }
+                }
+            }
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("using userIdClaimUriInOIDCDialect to get subject from idTokenClaims: " +
+                        userIdClaimUriInOIDCDialect);
+            }
+            Object subject = idTokenClaims.get(userIdClaimUriInOIDCDialect);
+            if (subject instanceof String) {
+                return (String) subject;
+            } else if (subject != null) {
+                LOG.warn("Unable to map subject claim (non-String type): " + subject);
+            }
+        } catch (ClaimMetadataException ex) {
+            throw new AuthenticationFailedException(
+                    UAEPassAuthenticatorConstants.ErrorMessages.EXECUTING_CLAIM_TRANSFORMATION_FOR_IDP_FAILED.getCode(),
+                    String.format(UAEPassAuthenticatorConstants.ErrorMessages.EXECUTING_CLAIM_TRANSFORMATION_FOR_IDP_FAILED.
+                                    getMessage(), context.getExternalIdP().getIdPName()), ex);
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Couldn't find the subject claim among id_token claims for IDP: " + context.getExternalIdP()
+                    .getIdPName());
+        }
+        return null;
+    }
+
+
+    /**
+     * Returns the claim URIs from UAEPass
+     *
+     * @param userIdClaimInLocalDialect
+     * @param spTenantDomain
+     * @return
+     * @throws ClaimMetadataException
+     */
+    public String getUserIdClaimUriInOIDCDialect(String userIdClaimInLocalDialect, String spTenantDomain)
+            throws ClaimMetadataException {
+
+        List<ExternalClaim> externalClaims = UAEPassDataHolder.getInstance()
+                .getClaimMetadataManagementService().getExternalClaims(UAEPassAuthenticatorConstants.
+                                UAEPassRuntimeConstants.OIDC_DIALECT, spTenantDomain);
+        String userIdClaimUri = null;
+        ExternalClaim oidcUserIdClaim = null;
+
+        for (ExternalClaim externalClaim : externalClaims) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(
+                        "Evaluating " + userIdClaimInLocalDialect + " against " + externalClaim.getMappedLocalClaim());
+            }
+            if (userIdClaimInLocalDialect.equals(externalClaim.getMappedLocalClaim())) {
+                oidcUserIdClaim = externalClaim;
+            }
+        }
+        if (oidcUserIdClaim != null) {
+            userIdClaimUri = oidcUserIdClaim.getClaimURI();
+        }
+        return userIdClaimUri;
```

Comment:
```suggestion

        return userIdClaimUri;
```
Missing new line before return statement. Check other places as well